src/main/scala/firrtl/Emitter.scala

```diff
@@ -382,16 +383,26 @@ class VerilogEmitter extends Emitter with PassBased {
         Seq("$fwrite(32'h80000002,", strx, ");")
       }
 
-      def build_ports(): Unit = portdefs ++= m.ports.zipWithIndex map {
-        case (p, i) => (p.tpe, p.direction) match {
-          case (AnalogType(_), _) =>
-            Seq("inout", "  ", p.tpe, " ", p.name)
-          case (_, Input) =>
-            Seq(p.direction, "  ", p.tpe, " ", p.name)
-          case (_, Output) =>
-            val ex = WRef(p.name, p.tpe, PortKind, FEMALE)
-            assign(ex, netlist(ex))
-            Seq(p.direction, " ", p.tpe, " ", p.name)
+      def build_ports(): Unit = {
+        def padToMax(strs: Seq[String]): Seq[String] = {
+          val len = if (strs.nonEmpty) strs.map(_.length).max else 0
+          strs map (_.padTo(len, ' '))
+        }
+        val dirs = m.ports map { case Port(_, name, dir, tpe) =>
+          (dir, tpe) match {
+            case (_, AnalogType(_)) => "inout"
+            case (Input, _) => stringify(dir)
+            case (Output, _) =>
+              // Assign to the Port
+              val ex = WRef(name, tpe, PortKind, FEMALE)
+              assign(ex, netlist(ex))
+              stringify(dir)
+          }
+        }
+        val tpes = m.ports map (p => stringify(p.tpe.asInstanceOf[GroundType]))
```

Comment:
Is this a reasonable cast to make? Should we check something first? Before the refactor it didn't need to make a cast.